texar/data/data/multi_aligned_data.py

```diff
@@ -195,22 +191,23 @@ def __init__(self, hparams, device: Optional[torch.device] = None):
         self._embedding = self.make_embedding(self._hparams.datasets,
                                               self._vocab)
 
-        self._dataset_features: List[Optional[Dict[str, Any]]] = []
         self._name_prefix: List[str] = []
+        self._names: List[Dict[str, Any]] = []
         datasources: List[DataSource] = []
         filters: List[Optional[Callable[[str], bool]]] = []
-        for _, hparams_i in enumerate(self._hparams.datasets):
+        self._databases: List[DataBase] = []
+        for idx, hparams_i in enumerate(self._hparams.datasets):
             dtype = hparams_i.data_type
             datasource_i: DataSource
+
             if _is_text_data(dtype) or _is_scalar_data(dtype):
                 datasource_i = TextLineDataSource(
                     hparams_i.files,
                     compression_type=hparams_i.compression_type)
                 datasources.append(datasource_i)
-                self._dataset_features.append(None)
-                if _is_text_data(dtype) and hparams_i["length_filter_mode"] is \
-                        _LengthFilterMode.DISCARD.value and \
-                        hparams_i["max_seq_length"] is not None:
+                if _is_text_data(dtype) and (hparams_i["length_filter_mode"] is
+                                             _LengthFilterMode.DISCARD.value) \
+                        and (hparams_i["max_seq_length"] is not None):
 
                     def _get_filter(delimiter, max_seq_length):
                         return lambda x: len(x.split(delimiter)) <= \
```

Comment:
The `x` in this function is the tuple of raw examples from all sources, but the check should only be applied on the `idx`-th element of the tuple.

Reply:
Not really, right? The `filter_fn` will receive tuple of raw examples from all sources, the `x` in individual filters is not a tuple, it is a component of the tuple on which filters have to be applied directly.

Reply:
Ah, I see, you're right. Sorry for reading the code incorrectly.